java/vector/src/main/codegen/templates/NullableValueVectors.java

```diff
@@ -122,7 +122,9 @@ public void initializeChildrenFromFields(List<Field> children) {
   public void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers) {
     <#if type.major = "VarLen">
     // variable width values: truncate offset vector buffer to size (#1)
-    org.apache.arrow.vector.BaseDataValueVector.truncateBufferBasedOnSize(ownBuffers, 1, values.offsetVector.getBufferSizeFor(fieldNode.getLength() + 1));
+    org.apache.arrow.vector.BaseDataValueVector.truncateBufferBasedOnSize(ownBuffers, 1,
+        values.offsetVector.getBufferSizeFor(
+        fieldNode.getLength() == 0? 0 : fieldNode.getLength() + 1));
```

Comment:
OK. I don't think this does any harm, but if there's a more appropriate fix it would be good to do that (there was another place where length-0 was being special-cased already)